src/app/file/gif_format.cpp

```diff
@@ -508,161 +512,136 @@ class GifDecoder {
     // We'll calculate the list of used colormap indexes in this
     // frameImage.
     PalettePicks usedEntries(ncolors);
-    if (isLocalColormap) {
-      // With this we avoid discarding the transparent index when a
-      // frame indicates that it uses a specific index as transparent
-      // but the image is completely opaque anyway.
-      if (!m_opaque && m_frameNum == 0 && m_localTransparentIndex >= 0 &&
-          m_localTransparentIndex < ncolors) {
-        usedEntries[m_localTransparentIndex] = true;
-      }
+    if (m_frameNum == 0 && !m_opaque &&
+        m_localTransparentIndex < ncolors &&
+        m_localTransparentIndex >= 0) {
+      m_bgIndex = m_localTransparentIndex;
+      m_sprite->setTransparentColor(m_bgIndex);
+      clear_image(m_currentImage.get(), m_bgIndex);
+      clear_image(m_previousImage.get(), m_bgIndex);
+      usedEntries[m_bgIndex] = true;
+    }
```

Comment:
Probably I'd add a comment about this (why the m_bgIndex changed to the local transparent index), as the old comment was lost in the change.